src/ast/passes/semantic_analyser.cpp

```diff
@@ -3155,6 +3155,13 @@ void SemanticAnalyser::visit(AttachPoint &ap)
         has_end_probe_ = true;
       }
     }
+  } else if (ap.provider == "signal") {
+    if (ap.target == "")
+      LOG(ERROR, ap.loc, err_) << ap.provider << " should have a target";
+    if (ap.func != "")
+      LOG(ERROR, ap.loc, err_) << ap.provider << " should only have a target";
```

Comment:
These should be caught by `AttachpointParser`, shouldn't they?

Reply:
oh right, checking the number of arguments in AttachpointParser seems enough.